src/Components/Components/src/Forms/EditForm.cs

```diff
@@ -0,0 +1,150 @@
+        [Parameter] EditContext EditContext { get; set; }
+
+        /// <summary>
+        /// Specifies the top-level model object for the form. An edit context will
+        /// be constructed for this model. If using this parameter, do not also supply
+        /// a value for <see cref="EditContext"/>.
+        /// </summary>
+        [Parameter] object Model { get; set; }
+
+        /// <summary>
+        /// Specifies the content to be rendered inside this <see cref="EditForm"/>.
+        /// </summary>
+        [Parameter] RenderFragment<EditContext> ChildContent { get; set; }
+
+        /// <summary>
+        /// A callback that will be invoked when the form is submitted.
+        ///
+        /// If using this parameter, you are responsible for triggering any validation
+        /// manually, e.g., by calling <see cref="EditContext.Validate"/>.
+        /// </summary>
+        [Parameter] Func<EditContext, Task> OnSubmit { get; set; }
+
+        /// <summary>
+        /// A callback that will be invoked when the form is submitted and the
+        /// <see cref="EditContext"/> is determined to be valid.
+        /// </summary>
+        [Parameter] Func<EditContext, Task> OnValidSubmit { get; set; }
+
+        /// <summary>
+        /// A callback that will be invoked when the form is submitted and the
+        /// <see cref="EditContext"/> is determined to be invalid.
+        /// </summary>
+        [Parameter] Func<EditContext, Task> OnInvalidSubmit { get; set; }
+
+        /// <inheritdoc />
+        protected override void OnParametersSet()
+        {
+            if ((EditContext == null) == (Model == null))
+            {
+                throw new InvalidOperationException($"{nameof(EditForm)} requires a {nameof(Model)} " +
+                    $"parameter, or an {nameof(EditContext)} parameter, but not both.");
+            }
+
+            // If you're using OnSubmit, it becomes your responsibility to trigger validation manually
+            // (e.g., so you can display a "pending" state in the UI). In that case you don't want the
+            // system to trigger a second validation implicitly, so don't combine it with the simplified
+            // OnValidSubmit/OnInvalidSubmit handlers.
+            if (OnSubmit != null && (OnValidSubmit != null || OnInvalidSubmit != null))
+            {
+                throw new InvalidOperationException($"When supplying an {nameof(OnSubmit)} parameter to " +
+                    $"{nameof(EditForm)}, do not also supply {nameof(OnValidSubmit)} or {nameof(OnInvalidSubmit)}.");
+            }
+
+            // Update _fixedEditContext if we don't have one yet, or if they are supplying a
+            // potentially new EditContext, or if they are supplying a different Model
+            if (_fixedEditContext == null || EditContext != null || Model != _fixedEditContext.Model)
+            {
+                _fixedEditContext = EditContext ?? new EditContext(Model);
+            }
```

Comment:
I'm not super clear on what this is for - the comment says what it does, but not why it's important.

Reply:
Looks like it's explained below.